cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/task/dnszone/AzureDnsZoneCreationCheckerTask.java

```diff
@@ -42,10 +42,10 @@ protected Boolean doCall() {
         ResourceStatus templateDeploymentStatus = azureClient.getTemplateDeploymentStatus(resourceGroupName, deploymentName);
 
         if (templateDeploymentStatus == ResourceStatus.DELETED) {
-            throw new CloudConnectorException(String.format("Deployment %s is either deleted of does not exist", deploymentName));
+            throw new CloudConnectorException(String.format("Deployment %s is either deleted or does not exist", deploymentName));
```

Comment:
Another question: in case there was no permission to create a DNS zone, how come this error message was shown to the user? How did the deployment get deleted?

Reply:
Because this exception got propagated to the UI instead of the Azure exception, before the fix